File: C++/1475. Final Prices With a Special Discount in a Shop.cpp
class Solution {
public:
    vector<int> finalPrices(vector<int>& p) {
        vector<int>a=p;
        vector<int>b;
        for(int i=0;i<p.size();i++){
            while(!b.empty() && p[b.back()]>=p[i]){
                a[b.back()]-=p[i];
                b.pop_back();
            }
            b.push_back(i);
        }
        return a;
    }
};

File: JAVA/1475. Final Prices With a Special Discount in a Shop.java
class Solution {
    public int[] finalPrices(int[] p) {
        int[] a=p.clone();
        Stack<Integer>b=new Stack<>();
        for(int i=0;i<p.length;i++){
            while(b.size()!=0 && p[b.peek()]>=p[i])
            a[b.pop()]-=p[i];
        b.add(i);
        }
    return a;
    }
}